common/fixation_entry.go

```diff
@@ -169,42 +166,66 @@ func (fs *FixationStore) AppendEntry(
 		Block:    block,
 		StaleAt:  math.MaxUint64,
 		Data:     marshaledEntryData,
-		Refcount: 0,
+		Refcount: 1,
 	}
 
 	fs.setEntry(ctx, entry)
-
-	fs.deleteStaleEntries(ctx, safeIndex)
-
 	return nil
 }
 
 func (fs *FixationStore) deleteStaleEntries(ctx sdk.Context, safeIndex string) {
 	fs.assertSanitizedIndex(safeIndex)
 	store := fs.getStore(ctx, safeIndex)
+
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
 	defer iterator.Close()
 
-	for iterator.Valid() {
-		// umarshal the old entry version
-		var oldEntry types.Entry
-		fs.cdc.MustUnmarshal(iterator.Value(), &oldEntry)
+	// remove "eligible" stale entry versions, where "eligible" is either:
+	//   an entry version that follows a stale entry version, -OR-
+	//   the older entry version
+	// rationale: we keep certain stale entry versions in order to mark -
+	// for future findEntry - that blocks from that stale entry onward are
+	// stale (otherwise, a lookup might resolve successfully with an older
+	// non-stale entry version). For this, one - the oldest - marker is
```

Comment:
why would we need the markers to mark which entry is stale? the refCount + STALE_TIME mechanism does just that. Also, what if we want to keep the oldest version of the entry? In other words, why is the oldest entry automatically stale?